libraries/chain/backing_store/tests/test_kv_rocksdb.cpp

```diff
@@ -0,0 +1,689 @@
+
+   }
+
+   BOOST_FIXTURE_TEST_CASE(test_iter_creation_key_size_too_big, kv_rocksdb_fixture)
+   {
+      BOOST_CHECK_NO_THROW(my_kv_context->kv_it_create(contract, prefix.c_str(), max_key_size));  // within the size
+      BOOST_CHECK_THROW(my_kv_context->kv_it_create(contract, prefix.c_str(), max_key_size + 1), kv_bad_iter); // outside of the max key size
+
+   }
+
+   BOOST_FIXTURE_TEST_CASE(test_is_kv_chainbase_context_iterator, kv_rocksdb_fixture)
+   {
+      std::unique_ptr<kv_iterator> it = my_kv_context->kv_it_create(contract, prefix.c_str(), it_key_size);
+      BOOST_CHECK(!it->is_kv_chainbase_context_iterator());
+   }
+
+   BOOST_FIXTURE_TEST_CASE(test_is_kv_rocksdb_context_iterator, kv_rocksdb_fixture)
+   {
+      std::unique_ptr<kv_iterator> it = my_kv_context->kv_it_create(contract, prefix.c_str(), it_key_size);
+      BOOST_CHECK(it->is_kv_rocksdb_context_iterator());
+   }
+
+   BOOST_FIXTURE_TEST_CASE(test_kv_it_compare, kv_rocksdb_fixture)
+   {
+      check_kv_it_compare(false, false, contract);
+   }
+
+   BOOST_FIXTURE_TEST_CASE(test_kv_it_compare_different_contract, kv_rocksdb_fixture)
+   {
+      check_kv_it_compare(false, true, contract + 1); // +1 to make contract different
+   }
+
+   BOOST_FIXTURE_TEST_CASE(test_kv_it_compare_erased, kv_rocksdb_fixture)
+   {
+      check_kv_it_compare(true, true, contract);
+   }
+
+   BOOST_FIXTURE_TEST_CASE(test_kv_it_key_compare_equal, kv_rocksdb_fixture)
+   {
+      check_kv_it_key_compare(false, "key", true);
+   }
+
+   BOOST_FIXTURE_TEST_CASE(test_kv_it_key_compare_non_equal, kv_rocksdb_fixture)
+   {
+      check_kv_it_key_compare(false, "randomkey", false);
+   }
+
+   BOOST_FIXTURE_TEST_CASE(test_kv_it_key_compare_erased, kv_rocksdb_fixture)
+   {
+      check_kv_it_key_compare(true, "key", true); // no use for last 2 argument when is_erased is true
+   }
+
+   BOOST_FIXTURE_TEST_CASE(test_kv_it_status_end, kv_rocksdb_fixture)
+   {
+      check_test_kv_it_status(true /* is_end */, true /* is_erased */, kv_it_stat::iterator_end);
+      check_test_kv_it_status(true, false, kv_it_stat::iterator_end);
+   }
+
+   BOOST_FIXTURE_TEST_CASE(test_kv_it_status_erased, kv_rocksdb_fixture)
+   {
+      check_test_kv_it_status(false /* is_end */, true /* is_erased */, kv_it_stat::iterator_erased);
+   }
+
+   BOOST_FIXTURE_TEST_CASE(test_kv_it_status_ok, kv_rocksdb_fixture)
+   {
+      check_test_kv_it_status(false, false, kv_it_stat::iterator_ok);
+   }
+
+   BOOST_FIXTURE_TEST_CASE(test_kv_it_move_to_end, kv_rocksdb_fixture)
+   {
+      mock_is_erased = []( ) -> bool {
+         return false;
+      };
+      std::unique_ptr<kv_iterator> it = my_kv_context->kv_it_create(contract, prefix.c_str(), it_key_size);
+      BOOST_CHECK(it->kv_it_move_to_end() == kv_it_stat::iterator_end);
+   }
+
+   BOOST_FIXTURE_TEST_CASE(test_kv_it_next_erased, kv_rocksdb_fixture)
+   {
+      check_kv_it_move(true, true, true, kv_it_stat::iterator_ok);
```

Comment:
The use of `bool`'s here makes it hard to really understand what these test cases are doing without going back and forth with the definition of `check_kv_it_*`

Maybe we can make a parameters structure so we can have some names aka:
```
enum it_dir {
   NEXT,
   PREV
}
struct check_kv_it_move_params {
   it_dir direction = NEXT;
   bool is_erased = false;
   bool is_end = false;
}

check_kv_it_move({}, kv_it_stat::iterator_ok);
check_kv_it_move({.is_end=true}, kv_it_stat::iterator_end);
check_kv_it_move({.direction=PREV, .is_erased=true, .is_end=true}, kv_it_stat::iterator_ok}
```

I'm not sold on the above approach but I think there is something that can be done for readability here.

Reply:
I was concern about those true/false sequences for readability too. Thanks for pointing this out. Even for test code we should strive for best code. Will modify the code.